src/main/java/no/unit/nva/hamcrest/DoesNotHaveNullOrEmptyFields.java

```diff
@@ -37,73 +39,71 @@ public void describeTo(Description description) {
 
     @Override
     public void describeMismatch(Object item, Description description) {
-        String emptyFieldNames = emptyFields
-            .stream()
-            .map(res -> res.methodName)
+        String emptyFieldNames = emptyFields.stream()
+            .map(res -> res.propertyName)
             .collect(Collectors.joining(
                 FIELD_DELIMITER));
 
         description.appendText("The following fields were found empty:")
             .appendText(emptyFieldNames);
     }
 
-    private boolean assertThatNoPublicFieldIsNull(Object insertedUser) {
-        Method[] methods = insertedUser.getClass().getMethods();
-        Stream<MethodInvocationResult> getterInvocations = Arrays.stream(methods)
-            .filter(this::isAGetter)
-            .map((method -> invokeMethodWithRuntimeException(insertedUser, method)));
-
-        List<MethodInvocationResult> emptyFields = getterInvocations.filter(this::isEmpty).collect(
-            Collectors.toList());
-        this.emptyFields = emptyFields;
+    private boolean assertThatNoPublicFieldIsNull(Object input) {
+        Stream<PropertyDescriptor> properties = retrieveProperties(input);
+        emptyFields = properties
+            .map(prop -> readPropertyValue(prop, input))
+            .filter(this::isEmpty)
+            .collect(Collectors.toList());
         return emptyFields.isEmpty();
     }
 
-    private MethodInvocationResult invokeMethodWithRuntimeException(Object insertedUser, Method method) {
+    private PropertyValuePair readPropertyValue(PropertyDescriptor prop, Object input) {
         try {
-            return invokeMethod(insertedUser, method);
+            Method getter = prop.getReadMethod();
+            Object propertyValue = getter.invoke(input);
+            return new PropertyValuePair(prop.getName(), propertyValue);
         } catch (IllegalAccessException | InvocationTargetException e) {
-            throw new RuntimeException(e);
+            throw new RuntimeException(PROPERTY_READ_ERRROR + prop.getName());
         }
     }
 
-    private MethodInvocationResult invokeMethod(Object insertedUser, Method method)
-        throws IllegalAccessException, InvocationTargetException {
-        Object result = method.invoke(insertedUser);
-        return new MethodInvocationResult(method.getName(), result);
-    }
-
-    private boolean isAGetter(Method m) {
-        return m.getName().startsWith(GETTER_GET_PREFIX) || m.getName().startsWith(GETTER_IS_PREFIX);
+    private Stream<PropertyDescriptor> retrieveProperties(Object input) {
+        try {
+            return Arrays.stream(Introspector
+                .getBeanInfo(input.getClass(), Object.class)
+                .getPropertyDescriptors());
+        } catch (IntrospectionException e) {
+            throw new RuntimeException(e);
+        }
     }
 
-    private static class MethodInvocationResult {
+    private static class PropertyValuePair {
```

Comment:
Could potentially use javafx.util.Pair

Reply:
I do not want to import and maintain another library for such a simple case.